### PR TITLE
build: feature gate arch and cpu specific code

### PR DESCRIPTION
**Pull request by mcoffin**
This will make zinc compatible for compilation for the build system.

Eventually, we should move architecture specific code somewhere, but this will suffice until then.

### CODE REVIEW

File: src/hal/mod.rs [hunk 1]
```diff
@@ -22,14 +22,23 @@ and each such struct has a `setup()` method that configures the hardware
 */
 
 
+#[cfg(feature = "mcu_lpc17xx")]
```
Comment (farcaller) [inline src/hal/mod.rs]: That's actually a bad idea I guess, as it feature-gates mcu code from being unit tested.

Comment (mcoffin) [inline src/hal/mod.rs]: But wouldn't you then just test with the features enabled (which we already do)? As a bonus it has exposed some misuse of hal code.

Comment (farcaller) [inline src/hal/mod.rs]: I mean — test the specific parts of the mcu hal, like doing a `volatilecell` replay verification for uart setup.

Comment (farcaller) [inline src/hal/mod.rs]: Wait, your comment actually makes sense to me now.

:+1:

**Comment by mcoffin** [comment]
Fixes #303

**Comment by mcoffin** [comment]
It would be nice if rustc would pass `target_cpu` as a cfg attribute, but it does not. might consider asking upstream if they'll add it.

**Comment by mcoffin** [comment]
Finally back after finals. Is this good to go?

**Comment by mcoffin** [comment]
retest this please

**Comment by farcaller** [comment]
And this is why you don't use socat to proxy the requests into vm. LGTM though.

**Pull request merged**